flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetInputFormat.java

```diff
@@ -0,0 +1,248 @@
+	 * @param messageType schema of parquet file
+	 */
+
+	protected ParquetInputFormat(Path path, MessageType messageType) {
+		super(path);
+		this.expectedFileSchema = checkNotNull(messageType, "messageType");
+		RowTypeInfo rowTypeInfo = (RowTypeInfo) ParquetSchemaConverter.fromParquetType(expectedFileSchema);
+		this.fieldTypes = rowTypeInfo.getFieldTypes();
+		this.fieldNames = rowTypeInfo.getFieldNames();
+		// read whole parquet file as one file split
+		this.unsplittable = true;
+	}
+
+	@Override
+	public void configure(Configuration parameters) {
+		super.configure(parameters);
+		parquetRecordReader.setSkipCorruptedRecord(this.skipCorruptedRecord);
+	}
+
+	public void selectFields(String[] fieldNames) {
+		checkNotNull(fieldNames, "fieldNames");
+		this.fieldNames = fieldNames;
+		RowTypeInfo rowTypeInfo = (RowTypeInfo) ParquetSchemaConverter.fromParquetType(expectedFileSchema);
+		TypeInformation[] selectFieldTypes = new TypeInformation[fieldNames.length];
+		for (int i = 0; i < fieldNames.length; i++) {
+			selectFieldTypes[i] = rowTypeInfo.getTypeAt(fieldNames[i]);
+		}
+		this.fieldTypes = selectFieldTypes;
+	}
+
+	@Override
+	public Tuple2<Long, Long> getCurrentState() {
+		return new Tuple2<>(this.parquetRecordReader.getCurrentBlock(),
+			this.parquetRecordReader.getRecordInCurrentBlock());
+	}
+
+	@Override
+	public void open(FileInputSplit split) throws IOException {
+		org.apache.hadoop.conf.Configuration configuration = new org.apache.hadoop.conf.Configuration();
+		InputFile inputFile =
+			HadoopInputFile.fromPath(new org.apache.hadoop.fs.Path(split.getPath().toUri()), configuration);
+		ParquetReadOptions options = ParquetReadOptions.builder().build();
+		ParquetFileReader fileReader = new ParquetFileReader(inputFile, options);
+		MessageType schema = fileReader.getFileMetaData().getSchema();
+		MessageType readSchema = getReadSchema(schema);
+
+		this.skipThisSplit = false;
+		this.parquetRecordReader = new ParquetRecordReader<>(new RowReadSupport(), readSchema, FilterCompat.NOOP);
+		this.parquetRecordReader.initialize(fileReader, configuration);
+		if (this.recordConsumed == null) {
+			this.recordConsumed = getRuntimeContext().getMetricGroup().counter("parquet-records-consumed");
+		}
+
+		LOG.debug(String.format("Open ParquetInputFormat with FileInputSplit [%s]", split.getPath().toString()));
+		if (skipThisSplit) {
+			LOG.warn(String.format(
+				"Escaped the file split [%s] due to mismatch of file schema to expected result schema",
+				split.getPath().toString()));
+		}
+	}
+
+	@Override
+	public void reopen(FileInputSplit split, Tuple2<Long, Long> state) throws IOException {
+		Preconditions.checkNotNull(split, "reopen() cannot be called on a null split.");
+		Preconditions.checkNotNull(state, "reopen() cannot be called with a null initial state.");
+
+		try {
+			this.open(split);
+		} finally {
+			if (state.f0 != -1) {
+				// open and read util the record we were before
```

Comment:
rephrase comment: "seek to the read position in the split that we were at when the checkpoint was taken."